src/contracts/dependencies/AbstractToken.ts

```diff
@@ -7,4 +7,14 @@ export abstract class AbstractToken extends Contract {
     const result = await this.makeCall<string>('balanceOf', [who], block);
     return toBigNumber(result);
   }
+
+  public async getAllowance(owner: Address, spender: Address, block?: number) {
+    const result = await this.makeCall<string>('allowance', [owner, spender], block);
+    return toBigNumber(result);
+  }
+
+  public async getTotalSupply(block?: number) {
```

Comment:
Please add method annotations / documentation describing the parameters (even if redundant).